swap/src/bin/swap.rs

```diff
@@ -358,6 +360,18 @@ where
         let minimum_amount = bid_quote.min_quantity;
         let maximum_amount = bid_quote.max_quantity;
 
+        let code = QrCode::new(deposit_address.to_string())?;
+        let qr_code = code
+            .render::<unicode::Dense1x2>()
+            .dark_color(unicode::Dense1x2::Light)
+            .light_color(unicode::Dense1x2::Dark)
+            .build();
+
+        eprintln!("QR Code:{}", qr_code.lines().next().unwrap());
+        for line in qr_code.lines().skip(1) {
+            eprintln!("        {}", line)
+        }
```

Comment:
I would like this to be skipped if the user passed `--json`, otherwise the QR code is going to break parsing the logs. The `--json` flag is meant for users who want to integrate the CLI into existing applications. For those, the expectation is that every line on stderr is valid JSON.

While you are at, can we extract this into a function such that we can write something like this:

```rust
if !json {
	eprintln!("{}", qr_code(deposit_address));
}
```

I would like `qr_code` to not have any side-effects and just return a String. That should be doable if you just join all the lines together with `\n`.

Thank you :)

Reply:
Done.